src/v/cluster/rm_stm.cc

```diff
@@ -671,6 +671,60 @@ ss::future<> rm_stm::stop() {
     return raft::state_machine::stop();
 }
 
+rm_stm::transaction_info::status_t
+rm_stm::get_tx_status(model::producer_identity pid) {
+    if (_mem_state.preparing.find(pid) != _mem_state.preparing.end()) {
+        return transaction_info::preparing;
+    }
+
+    if (_log_state.prepared.find(pid) != _log_state.prepared.end()) {
+        return transaction_info::prepared;
+    }
+
+    return transaction_info::ongoing;
+}
+
+std::optional<rm_stm::expiration_info>
+rm_stm::get_expiration_info(model::producer_identity pid) {
+    auto it = _mem_state.expiration.find(pid);
+    if (it == _mem_state.expiration.end()) {
+        return std::nullopt;
+    }
+
+    return it->second;
+}
+
+absl::btree_map<model::producer_identity, rm_stm::transaction_info>
+rm_stm::get_transactions() {
+    absl::btree_map<model::producer_identity, rm_stm::transaction_info> ans;
```

Comment:
`_mem_state` and `_log_state` contain valid data only if the current node is a leader so before using the structures we should enforce it with 

```
if (!co_await sync(_sync_timeout)) {
    co_return errc::not_leader;
}
```